ansible_sdk/_aiocompat/file_async.py

```diff
@@ -0,0 +1,22 @@
+# Copyright: Ansible Project
+# Apache License 2.0 (see LICENSE or https://www.apache.org/licenses/LICENSE-2.0)
+
+import asyncio
+from typing import Any, Dict, Optional
+
+
+class AsyncFile:
+    def __init__(self, file_path: str, mode: str = "r", **kwargs: Optional[Dict[str, Any]]):
+        self.file_path = file_path
+        self.mode = mode
+        self.kwargs = kwargs
+        self.file = None
+
+    async def __aenter__(self):
+        self.file = await asyncio.to_thread(
```

Comment:
It's not the most efficient way, but try adding another one of these right after that just calls `self.file.__enter__`